### PR TITLE
Migrated matching engine notebook to official and added VPC network support to CI

### PR DESCRIPTION
**Pull request by ivanmkc**
Migrated matching engine notebook to official and added VPC network support to CI.

If you are opening a PR for `Official Notebooks` under the [notebooks/official](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/notebooks/official) folder, follow this mandatory checklist:
- [x] Use the [notebook template](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/notebook_template.ipynb) as a starting point.
- [x] Follow the style and grammar rules outlined in the above notebook template.
- [ ] Verify the notebook runs successfully in Colab since the automated tests cannot guarantee this even when it passes.

This won't run in Colab due to VPC network requirement.
- [x] Passes all the required automated checks. You can locally test for formatting and linting with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).
- [x] You have consulted with a tech writer to see if tech writer review is necessary. If so, the notebook has been reviewed by a tech writer, and they have approved it.
- [x] This notebook has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/official/CODEOWNERS) file under the `Official Notebooks` section, pointing to the author or the author's team.
- [x] The Jupyter notebook cleans up any artifacts it has created (datasets, ML models, endpoints, etc) so as not to eat up unnecessary resources.

### CODE REVIEW

File: .cloud-build/execute_changed_notebooks_helper.py [hunk 1]
```diff
@@ -286,12 +297,13 @@ def process_and_execute_notebooks(
     container_uri: str,
     staging_bucket: str,
     artifacts_bucket: str,
+    should_parallelize: bool,
+    timeout: int,
     variable_project_id: str,
     variable_region: str,
     variable_service_account: str,
-    private_pool_id: Optional[str],
```
Comment (ivanmkc) [inline .cloud-build/execute_changed_notebooks_helper.py:292]: Fixed order of optional and non-optional args.

File: .cloud-build/execute_changed_notebooks_helper.py [hunk 2]
```diff
@@ -389,11 +403,18 @@ def process_and_execute_notebooks(
                         format_timedelta(result.duration),
                         result.log_url,
                         result.output_uri,
-                        result.output_uri_web
+                        result.output_uri_web,
                     ]
                     for result in results_sorted
                 ],
-                headers=["build_tag", "status", "duration", "log_url", "output_uri", "output_uri_web"],
+                headers=[
```
Comment (ivanmkc) [inline .cloud-build/execute_changed_notebooks_helper.py:410]: Linter fixed this.

File: .cloud-build/execute_changed_notebooks_cli.py [hunk 3]
```diff
@@ -114,10 +120,11 @@ def str2bool(v):
     container_uri=args.container_uri,
     staging_bucket=args.staging_bucket,
     artifacts_bucket=args.artifacts_bucket,
+    should_parallelize=args.should_parallelize,
```
Comment (ivanmkc) [inline .cloud-build/execute_changed_notebooks_cli.py:123]: Fixed arg order.

File: .cloud-build/execute_changed_notebooks_helper.py [hunk 4]
```diff
@@ -128,15 +130,23 @@ def process_and_execute_notebook(
     variable_project_id: str,
     variable_region: str,
     variable_service_account: str,
+    variable_vpc_network: Optional[str],
     private_pool_id: Optional[str],
-    deadline: datetime,
+    deadline: datetime.datetime,
```
Comment (ivanmkc) [inline .cloud-build/execute_changed_notebooks_helper.py:135]: Fixed wrong typehint.

File: notebooks/official/CODEOWNERS [hunk 5]
```diff
@@ -5,7 +5,7 @@
 * @GoogleCloudPlatform/vertex-ai-samples-contributors  @GoogleCloudPlatform/caiis-tw
 
 # matching_engine folder
-/matching_engine @shenzhimo2
+/matching_engine @shenzhimo2 @ivanmkc
```
Comment (ivanmkc) [inline notebooks/official/CODEOWNERS:8]: Will fix CODEOWNER errors in another PR.

**Comment by soheilazangeneh** [comment]
Do we have a notebook in our CI test notebooks list that uses the new feature?

**Comment by ivanmkc** [comment]
> Do we have a notebook in our CI test notebooks list that uses the new feature?

Added to the notebooks-vm list.

**Review by soheilazangeneh — APPROVED** [review]

**Pull request merged**